### ISSUE TITLE
Change Title of Backend Page

### ISSUE THREAD

**Issue by ddeimeke** [s9y/Serendipity#413]
Please change the title of the backend page (shown with BLOGURL/serendipity_admin.php) to something that reflects the blog title.

Should be different from "Startseite | Serendipity Verwaltungsoberfläche", maybe "BLOGTITLE | Admin"?

**Comment by yellowled** [comment]
I assume the idea behind this is to not confuse multiple s9y installations in browser tabs?

We would need to make `$admin_vars.title` be the blog title for the admin index page. Not sure where to do that, but that's the variable that backend templates use here.

**Comment by ddeimeke** [comment]
Yes, exactly @yellowled

**Comment by onli** [comment]
Currently, the scheme is $admin_section | Serendipity Verwaltungsoberfläche. We could change it to $admin_section | blogTitle | | Serendipity Verwaltungsoberfläche.

**Comment by ddeimeke** [comment]
Ideally the $blogTitle would be first. 

And "Serendipity Verwaltunspberfläche" could be shortenned to "Admin". Since tabs are not that wide, most people (including myself) do not see the whole title.

**Comment by ddeimeke** [comment]
I see that most of the section don't have $admin_section set.

As examples tags, static pages, Spamschutz has "Plugins verwalten", statistics, ...

**Comment by onli** [comment]
I don't think we would really want blog title  first. In most situations all tabs would then only contain "a part of the blogtitle…". It would make matters worse – I'm missing nothing?

> And "Serendipity Verwaltunspberfläche" could be shortenned to "Admin". Since tabs are not that wide, most people (including myself) do not see the whole title.

The window title should show the whole text though.

> As examples tags, static pages, Spamschutz has "Plugins verwalten", statistics, ...

Those are all pages generated by plugins. I think we have no mechanism to set a better title there.

**Comment by ddeimeke** [comment]
> I don't think we would really want blog title first. In most situations all tabs would then only contain "a part of the blogtitle…". It would make matters worse – I'm missing nothing?

My blog shows tabs with "Dirks Logbuch", showing the backend title as "Dirks Logbuch | Admin" would be a major improvement.

> The window title should show the whole text though.

I do not have a window title. And even if I had, the scenario is that I have for example tabs open with different blog backends and all I see as tab titles is "Startseite ..." (nothing more, that's all).

> Those are all pages generated by plugins. I think we have no mechanism to set a better title there.

If not all plugins set the $admin_section properly, it is not needed anymore in my opinion. 

To see what I mean, take your favorite browser, open ten tabs with the backend open (even with the same blog) and tell me if the title you see helps you to distinguish different sessions.

**Comment by onli** [comment]
Okay, let's play this through :) I think this is a nice small UX issue we should strive to solve as good as possible.

We have two goals here: Make different backend tabs of the same blog distinguishable (priority, imho), and make backends of different blogs distinguishable (will be less often important, but would still be nice).

> If not all plugins set the $admin_section properly, it is not needed anymore in my opinion

It is there to make different backend sections distinguishable, and apart from the plugin problem it is working fine.

> And even if I had [a title], the scenario is that I have for example tabs open with different blog backends and all I see as tab titles is "Startseite ..." (nothing more, that's all).

`echo "Startseite " | wc -c` => 12

I'll now open 5 backend pages in my two blogs and each time write down which tab title they would have. I marked in bold which part would be visible in your browser. 12 chars seems very small to me, I see like 20, i'll mark that italic.

Blog | Current scheme | Blogtitle - Section - Admin | Section - Blogtitle - Admin 
-------|-----------|----------|-------
onli-blogging | **Startseite -** *Serendi*pity Verwaltungsoberfläche | **onli-bloggin***g* *-* *Star*tseite - Serendipity Verwaltungsoberfläche | **Startseite -** *onli-blo*gging - Serendipity Verwaltungsoberfläche
devblog | **Startseite -** *Serendi*pity Verwaltungsoberfläche | **John Doe's P***ersonal* Blog - Startseite - Serendipity Verwaltungsoberfläche | **Startseite -** *John Do*e's Personal Blog - Serendipity Verwaltungsoberfläche
onli-blogging | **Einträge - S**e*rendipi*ty Verwaltungsoberfläche | **onli-bloggin***g* *-* *Ein*träge - Serendipity Verwaltungsoberfläche | **Einträge - o***nli-blog*ging - Serendipity Verwaltungsoberfläche
devblog | **Einträge - S***erendipi*ty Verwaltungsoberfläche | **John Doe's P***ersonal* Blog - Einträge - Serendipity Verwaltungsoberfläche | **Einträge - J***ohn Doe'*s Personal Blog - Serendipity Verwaltungsoberfläche
onli-blogging | **Plugins verw***alten* *-* Serendipity Verwaltungsoberfläche | **onli-bloggi***ng* *-* *Plu*gins verwalten - Serendipity Verwaltungsoberfläche | **Plugins ver***walten* *-* onli-blogging - Serendipity Verwaltungsoberfläche
devblog | **Plugins ver***walten* *-* Serendipity Verwaltungsoberfläche | **John Doe's P***ersonal* Blog - Plugins verwalten - Serendipity Verwaltungsoberfläche | **Plugins ver***walten* *-* John Doe's Personal Blog - Serendipity Verwaltungsoberfläche
onli-blogging | **Themes verwa***lten* *-* *S*erendipity Verwaltungsoberfläche | **onli-bloggi***ng* *-* *The*mes verwalten - Serendipity Verwaltungsoberfläche | **Themes verwa***lten* *-* *onl*i-blogging - Serendipity Verwaltungsoberfläche
devblog | **Themes verwa***lten* *-* *S*erendipity Verwaltungsoberfläche | **John Doe's P***ersonal* Blog - Themes verwalten - Serendipity Verwaltungsoberfläche | **Themes verwa***lten* *-* *J*ohn Doe's Personal Blog - Serendipity Verwaltungsoberfläche
onli-blogging | **Mediendatenb***ank* *-* *Se*rendipity Verwaltungsoberfläche | **onli-blogg***ing* *-* *Med*iendatenbank - Serendipity Verwaltungsoberfläche | **Mediendatenb***ank* *-* *on*li-blogging - Serendipity Verwaltungsoberfläche
devblog | **Mediendatenb***ank* *-* *Se*rendipity Verwaltungsoberfläche | **John Doe's P***ersonal* Blog - Mediendatenbank - Serendipity Verwaltungsoberfläche | **Mediendatenb***ank* *-* *Jo*hn Doe's Personal Blog - Serendipity Verwaltungsoberfläche

What a work of art. But I think it is understandable You see the problem: If we have the blog title first, a longer blog title will mean that in all tabs only the blog title is visible. Blogs would be distinguishable, but multiple tabs of the same blog would not.

In the current scheme, multiple tabs are well distinguishable, but blogs never are.

In the scheme I propose – Section | blogTitle | Admin - tabs of the same blog are always distinguishable, and which blog is open sometimes. We could improve that by making the section titles shorter.

However, the correct way to distinguish between different websites is not the title alone, it is the favicon. I see one in my devblog (for all sites on the server it is the same), but not in my main blog (the favicon there works only for the frontend). Can we make the favicon appear in the backend, @yellowled ?

**Comment by yellowled** [comment]
> I see one in my devblog (for all sites on the server it is the same), but not in my main blog (the favicon there works only for the frontend). Can we make the favicon appear in the backend, @yellowled ?

The favicon *should* appear in the backend if used “properly”, which as far as I know means either place it in the document root or link to it using the proper `<meta>` element in the `<head>`. I've never been a fan of using the meta element, and if we were to use it, we'd need an interface to pick the (path to the) favicon including other logic (like checking if it fulfills the proper format etc.)

So, bottom line, it *should* already work if people put their favicon in the document root of the server. Also, favicons *always* require an extra effort be the user to create proper favicons (which, granted, is not that hard; there's tons of online generators).

It's also worth noting that favicons are the *only* way to differentiate “pinned” tabs (not sure if all browsers have those, though).

**Comment by ddeimeke** [comment]
Sorry to say, but this
> In the current scheme, multiple tabs are well distinguishable, but blogs never are.
is not true for many reasons. Not all Plugins populate $admin_section properly and lead to non-distinguishable Tabs.

> Make different backend tabs of the same blog distinguishable (priority, imho)

Not for me ... I maybe seldomly have two tabs of the same blog open, but in an update session I have numerous backends of different blogs open.

"Section | blogTitle | Admin" leads to the same problems I already have. If section is properly filled, we don't need admin anymore this is redundant. 

Having a favicon is a bonus but not a must.

**Comment by onli** [comment]
Thanks @yellowled. Maybe I'd have to add a favicon.ico instead of favicon.png, or something like that (it is in the document root). 

Maybe we could find a nice solution and generate a random favicon and set it via the meta tag in the backend?

> "Section | blogTitle | Admin" leads to the same problems I already have. If section is properly filled, we don't need admin anymore this is redundant. 

Section is something short like "Entries". It does not make clear that it is the admin backend that is open, and that can't be added to section or it would become too long again.

>> Make different backend tabs of the same blog distinguishable (priority, imho)
>
> Not for me ... I maybe seldomly have two tabs of the same blog open, but in an update session I have numerous backends of different blogs open

I don't see a better solution here. My proposed solution is the one possible change I see that gives you at least a chance of distinguishing between blogs, more so if we succeed in making the section titles shorter and/or if you succeed in making the tabtitle in your browser longer (something like the treestyle tab plugin might be useful for your use case).

 Having blogTitle first is bad for every user that opens more than one tab of his blog and only uses one at a time, which will be the big majority. I think that is no acceptable drawback.

**Comment by yellowled** [comment]
> Maybe I'd have to add a favicon.ico instead of favicon.png, or something like that (it is in the document root).

Probably, yes. These things are ridiculously complicated because as always, different browser vendors handled things differently. Got even worse with the addition of [touch icons](https://mathiasbynens.be/notes/touch-icons). Might also have something to do with what exactly the web server considers to be document root (which I guess makes it even harder to have a standard solution that works for everyone).

> Maybe we could find a nice solution and generate a random favicon and set it via the meta tag in the backend?

Random? Why not use the ones used in https://github.com/s9y/s9y.github.io …? But those would definitely require using the `<meta>` tag for *both* icons.

```
<link rel="icon" href="/path/to/favicon.ico" type="image/x-icon">
<link rel="apple-touch-icon" href="/path/to/apple-touch-icon.png">
```

**Comment by onli** [comment]
> Random? Why not use the ones used in https://github.com/s9y/s9y.github.io …?

Then they would not by default differ between two different blogs.

Tried that with the .ico, doesn't work yet, indeed complicated. Well, a different issue and not s9y specific.

**Comment by yellowled** [comment]
Yeah, but random ones do not exactly make it easier to identify which one's which. I think this is a lost cause.

**Comment by ddeimeke** [comment]
It could be pretty easy to make this configurable and offer three possible solutions to choose from.

In my opinion the majority of people only use one tab or don't even care what the title is.

**Comment by onli** [comment]
> Yeah, but random ones do not exactly make it easier to identify which one's which

Not random on each load, random once on install/upgrade, to have an unique identifier.

> It could be pretty easy to make this configurable and offer three possible solutions to choose from.

I think that is a good example of something that should not be configurable. I don't manage to translate that properly – Das wäre doch viel zu kleinteilig, und für zu viele Nutzer nicht interessant. 

Proposal: I'll push the section | blogTitle | admin solution. Then we look at making the longer section titles shorter. Let's also see whether the favicon idea can work out, I think it could be useful and doable. If we are there and you still have the same problem, then we look again at alternative solutions, including making it configurable.

**Comment by garvinhicking** [comment]
We would like to add a configuration option (inside config_local) to make the user decide if he wants to have:

[section] | [blogname]
OR:
[blogname] | [section]
